bdew_datetimes/german_strom_and_gas_tag.py

```diff
@@ -0,0 +1,103 @@
+"""
+A module to evaluate datetimes and whether they are "on the edge"
+of a German "Stromtag" or "Gastag" respectively
+"""
+from datetime import datetime, time
+from enum import Enum
+from typing import Callable
+
+# The problem with the stdlib zoneinfo is, that the availability of timezones
+# via ZoneInfo(zone_key) depends on the OS and system on which you're running
+# it. In some cases "Europe/Berlin" might be available, but generally it's not,
+# and it's PITA to manually define timezones. So we're using pytz as a
+# datasource for timezone information.
+from pytz import timezone, utc
+
+berlin = timezone("Europe/Berlin")
+
+
+class Division(Enum):
+    """
+    allows to distinguish divisions used by German utilities, German "Sparte"
+    """
+
+    STROM = (1,)  #: electricity
+    GAS = 2  #: gas
+
+
+def _get_german_local_time(date_time: datetime) -> time:
+    """
+    returns the current german local time for the given datetime object
+    """
+    german_local_datetime = date_time.astimezone(berlin)
+    return german_local_datetime.time()
+
+
+def has_no_utc_offset(date_time: datetime) -> bool:
+    """
+    returns true if and only if date_time has an explicit offset
+    which and the UTC-offset is 0. This means the UTC offset is exactly "+00:00".
+    """
+    # the name of the function contains a negation because in German
+    # market communication it often matters that the UTC offset is 0.
+    original_time = date_time.time()
+    utc_time = date_time.astimezone(tz=utc).time()
+    return (
+        utc_time == original_time
+        and utc_time.hour == 0
+        and utc_time.minute == 0
+        and utc_time.second == 0
+    )
+
+
+def is_stromtag_limit(
+    date_time: datetime,
+) -> bool:  # the name is not as speaking as I'd like it to be
```

Comment:
Maybe is_power_utilities_bound  ?

Reply:
I've thought about somthing like market/business day, but this could be confusing as the delivery is continuous.